server/src/main/java/org/apache/iotdb/db/metadata/MTree.java

```diff
@@ -899,6 +914,70 @@ private void findNodes(MNode node, String path, List<String> res, int targetLeve
     }
   }
 
+  public int getSnapshotLineNumber() {
+    return snapshotLineNumber;
+  }
+
+  public void serializeTo(String snapshotPath, int lineNumber) throws IOException {
+    try (BufferedWriter bw = new BufferedWriter(
+        new FileWriter(SystemFileFactory.INSTANCE.getFile(snapshotPath)))) {
+      bw.write(String.valueOf(lineNumber));
+      bw.newLine();
+      root.serializeTo(bw);
+    }
+  }
+
+  public static MTree deserializeFrom(String mtreeSnapshotPath) throws IOException {
```

Comment:
catch all exception and return an empty MTree if meet a broken file

Reply:
Good point! Fixed.